File: server/services/battle/wall.dart
import 'dart:math';

import 'battle.dart';

class Wall {

  const Wall(this.level, this.durability);
  final int level;
  final int durability;

  WallBonus<double> getBonus(int lvl, double base) {
    final defBonus = Fns.roundP(0.001, pow(base, lvl).toDouble()) - 1.0;
    return WallBonus.off(defBonus, 0);
  }
}

File: java_server/src/main/java/xyz/qruto/java_server/models/battle/Wall.java
package xyz.qruto.java_server.models.battle;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Wall {
    private int level;
    private int durability;

    public WallBonus<Double> getBonus(int lvl, double base){
        var defBonus = Fns.roundP(0.001, Math.pow(base, lvl)) - 1;
        return WallBonus.off(defBonus, 0.0);
    }
}
